File: src/test/unit/TranscriptTest.java
package test.unit;

import static org.junit.Assert.assertEquals;

import java.util.ArrayList;
import java.util.List;

import org.junit.Test;

import Debt_LRA_Transcript.Debt;
import Debt_LRA_Transcript.Transcript;
import IDs.StudentID;
import lecture.Semester;
import person.Student;

public class TranscriptTest {
	
	@Test
	public void test_transcript_student() {
		StudentID testId = new StudentID(000,000,000);
		Semester testSemester = new Semester(null);
		Semester testSemester2 = new Semester(null);
		List<Semester> testListOfSemester = new ArrayList<>();
		testListOfSemester.add(testSemester);
		testListOfSemester.add(testSemester2);
		
		Transcript testTranscript = new Transcript(null, testListOfSemester);
		
		Student testStudent2 = new Student("testFName2", "testLName2",
				testId, null, testTranscript, null);
				
		testTranscript.setStudent(testStudent2);;
		assertEquals(testStudent2, testTranscript.getStudent());		
	}
	@Test
	public void test_transcript_list_of_semester() {
		Semester testSemester = new Semester(null);
		Semester testSemester2 = new Semester(null);
		List<Semester> testListOfSemester = new ArrayList<>();
		testListOfSemester.add(testSemester);
		testListOfSemester.add(testSemester2);
		
		Semester testSemester3 = new Semester(null);
		List<Semester> testListOfSemester2 = new ArrayList<>();
		testListOfSemester2.add(testSemester3);
		testListOfSemester2.add(testSemester2);
		
		Transcript testTranscript = new Transcript(null, testListOfSemester);
				
		testTranscript.setListOfSemester(testListOfSemester2);
		assertEquals(testListOfSemester2 , testTranscript.getListOfSemester());
	}
	@Test
	public void test_transcript_total_credits_completed() {
		Semester testSemester = new Semester(null);
		Semester testSemester2 = new Semester(null);
		List<Semester> testListOfSemester = new ArrayList<>();
		testListOfSemester.add(testSemester);
		testListOfSemester.add(testSemester2);
		
		Transcript testTranscript = new Transcript(null, testListOfSemester);
			
		testTranscript.setTotalCreditsCompleted(6);
		assertEquals(6 , testTranscript.getTotalCreditsCompleted());	
	}
	@Test
	public void test_transcript_gano() {
		Semester testSemester = new Semester(null);
		Semester testSemester2 = new Semester(null);
		List<Semester> testListOfSemester = new ArrayList<>();
		testListOfSemester.add(testSemester);
		testListOfSemester.add(testSemester2);
		
		Transcript testTranscript = new Transcript(null, testListOfSemester);
		
		testTranscript.setGano(3.2);
		assertEquals(3.2 , testTranscript.getGano(),0.001);
	}
	@Test
	public void test_transcript_points() {
		Semester testSemester = new Semester(null);
		Semester testSemester2 = new Semester(null);
		List<Semester> testListOfSemester = new ArrayList<>();
		testListOfSemester.add(testSemester);
		testListOfSemester.add(testSemester2);
		
		Transcript testTranscript = new Transcript(null, testListOfSemester);

		testTranscript.setPoints(5.1);
		assertEquals(5.1 , testTranscript.getPoints(),0.001);
	}
	@Test
	public void test_transcript_total_credits_taken() {
		Semester testSemester = new Semester(null);
		Semester testSemester2 = new Semester(null);
		List<Semester> testListOfSemester = new ArrayList<>();
		testListOfSemester.add(testSemester);
		testListOfSemester.add(testSemester2);	
		
		Transcript testTranscript = new Transcript(null, testListOfSemester);
		
		testTranscript.setTotalCreditsTaken(8);
		assertEquals(8, testTranscript.getTotalCreditsTaken());
	}
}

File: pythonCode/TranscriptTest.py
import unittest
from Transcript import Transcript
from Student import Student
from Lecture import Lecture
from Semester import Semester
class TranscriptTest(unittest.TestCase):
    # generated source for class TranscriptTest
    def test_transcript_student(self):
        # generated source for method test_transcript_student 
        testTranscript = Transcript()
        testStudent = Student()
        testStudent.setFirstName("testFName")
        testTranscript.setStudent(testStudent)
        self.assertEqual(testStudent, testTranscript.getStudent())
    def test_transcript_ListOfSemester(self):
        # generated source for method test_transcript_ListOfSemester 
        testTranscript = Transcript()
        testSemester = Semester()
        testListOfSemester = list(testSemester)
        testTranscript.setListOfSemester(testListOfSemester)
        self.assertEqual(testListOfSemester, testTranscript.getListOfSemester())
    def test_semester_Gano(self):
        # generated source for method test_semester_Gano 
        testTranscript = Transcript()
        testTranscript.setGano(3.2)
        self.assertEqual(3.2, testTranscript.getGano)
    def test_transcript_TotalCreditsTaken(self):
        # generated source for method test_transcript_TotalCreditsTaken 
        testTranscript = Transcript()
        testTranscript.setTotalCreditsTaken(5)
        self.assertEqual(5, testTranscript.getTotalCreditsTaken())
    def test_transcript_Points(self):
        # generated source for method test_transcript_Points 
        testTranscript = Transcript()
        testTranscript.setPoints(5)
        self.assertEqual(5, testTranscript.getPoints())
    def test_transcript_TotalCreditsCompleted(self):
        # generated source for method test_transcript_TotalCreditsCompleted 
        testTranscript = Transcript()
        testTranscript.setTotalCreditsCompleted(10)
        self.assertEqual(10, testTranscript.getTotalCreditsCompleted())
if __name__ == '__main__':
    unittest.main()